api/src/cli/commands/init/questions.ts

```diff
@@ -7,69 +7,17 @@ const filename = ({ filepath }: { filepath: string }): Record<string, string> =>
 	default: path.join(filepath, 'data.db'),
 });
 
-const host = (): Record<string, string> => ({
+const connection_string = (): Record<string, string> => ({
 	type: 'input',
-	name: 'host',
-	message: 'Database Host:',
-	default: '127.0.0.1',
-});
-
-const port = ({ client }: { client: string }): Record<string, any> => ({
-	type: 'input',
-	name: 'port',
-	message: 'Port:',
-	default() {
-		const ports: Record<string, number> = {
-			pg: 5432,
-			cockroachdb: 26257,
-			mysql: 3306,
-			oracledb: 1521,
-			mssql: 1433,
-		};
-
-		return ports[client];
-	},
-});
-
-const database = (): Record<string, string> => ({
-	type: 'input',
-	name: 'database',
-	message: 'Database Name:',
-	default: 'directus',
-});
-
-const user = (): Record<string, string> => ({
-	type: 'input',
-	name: 'user',
-	message: 'Database User:',
-});
-
-const password = (): Record<string, string> => ({
-	type: 'password',
-	name: 'password',
-	message: 'Database Password:',
-	mask: '*',
-});
-
-const encrypt = (): Record<string, string | boolean> => ({
-	type: 'confirm',
-	name: 'options__encrypt',
-	message: 'Encrypt Connection:',
-	default: false,
-});
-
-const ssl = (): Record<string, string | boolean> => ({
-	type: 'confirm',
-	name: 'ssl',
-	message: 'Enable SSL:',
-	default: false,
+	name: 'connection_string',
+	message: 'Database Connection String:',
 });
 
 export const databaseQuestions = {
 	sqlite3: [filename],
-	mysql: [host, port, database, user, password],
-	pg: [host, port, database, user, password, ssl],
-	cockroachdb: [host, port, database, user, password, ssl],
-	oracledb: [host, port, database, user, password],
-	mssql: [host, port, database, user, password, encrypt],
+	mysql: [connection_string],
+	pg: [connection_string],
+	cockroachdb: [connection_string],
+	oracledb: [connection_string],
+	mssql: [connection_string],
```

Comment:
Is it common for all these databases to rely on a connection string? From what I understand, Oracle relies on a "connect string" (instead of "connection string"), and for mysql/mssql is still more common to use parameters right? 🤔 

(Also lets use `camelCase` for variable names in the codebase 👍🏻 )

Reply:
I'd say using separate parameters is still very common among webdevelopers. Some accompanying docs would be needed to instruct users how to build such a connection string for engines that don't provide it on startup like MySQL.

Reply:
@rijkvanzanten Another thought, what if we kept the separate parameters...but the first parameter is Host or Connection URL and then we detect if it is a connection string skip the rest of the parameters?

Reply:
> @rijkvanzanten Another thought, what if we kept the separate parameters...but the first parameter is Host or Connection URL and then we detect if it is a connection string skip the rest of the parameters? Allows people to use both ways.

In that case, I'd rather start with a question:

```
Use:
(•) Connection string
( ) Individual Parameters
```

But tbh, it feels like we can have smart enough defaults by using connection string for postgres/cockroach/oracle, and parameters for everything else

Reply:
That makes more sense, however, I wonder how this will affect people used to the default values? With this change won't they have to make their own connection string and there will be no defaults?

Reply:
> Is it common for all these databases to rely on a connection string? From what I understand, Oracle relies on a "connect string" (instead of "connection string"), and for mysql/mssql is still more common to use parameters right? thinking
> 
> (Also lets use `camelCase` for variable names in the codebase 👍🏻 )

Yes you're right. I'll make those updates today :)

Reply:
> > @rijkvanzanten Another thought, what if we kept the separate parameters...but the first parameter is Host or Connection URL and then we detect if it is a connection string skip the rest of the parameters? Allows people to use both ways.
> 
> In that case, I'd rather start with a question:
> 
> ```
> Use:
> (•) Connection string
> ( ) Individual Parameters
> ```
> 
> But tbh, it feels like we can have smart enough defaults by using connection string for postgres/cockroach/oracle, and parameters for everything else

I'm for this. Supporting both connection params and individual params, by asking the question after the user select their DB client.